packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -286,10 +288,15 @@ export class QueuedRequestController extends BaseController<
     if (!this.#originOfCurrentBatch) {
       throw new Error('Current batch origin must be initialized first');
     }
-    const originNetworkClientId = this.messagingSystem.call(
-      'SelectedNetworkController:getNetworkClientIdForDomain',
+    const originChainId = this.messagingSystem.call(
+      'SelectedNetworkController:getChainIdForDomain',
       this.#originOfCurrentBatch,
     );
+    const originNetworkClientId = this.messagingSystem.call(
+      'NetworkController:getDefaultNetworkClientIdForChainId',
+      originChainId,
+    );
```

Comment:
Is this the bit you were saying was awkward? Yeah I see that. Not a big deal though. And we're going to be removing the need for these calls altogether very soon!